Sources/SwiftParser/StringLiterals.swift

```diff
@@ -44,6 +148,243 @@ extension Parser {
+          RawStringSegmentSyntax(
+            RawUnexpectedNodesSyntax(combining: lastMiddleSegment.unexpectedBeforeContent, unexpectedBeforeContent, arena: self.arena),
+            content: content,
+            lastMiddleSegment.unexpectedAfterContent,
+            arena: self.arena
+          )
+        )
+        closeDelimiterOnNewLine = true
+      } else {
+        closeDelimiterOnNewLine = false
+      }
+    case .expressionSegment:
+      closeDelimiterOnNewLine = false
+    case nil:
+      if let firstSegment = firstSegment {
+        closeDelimiterOnNewLine = firstSegment.content.tokenText.newlineSuffix != nil
+      } else {
+        closeDelimiterOnNewLine = false
+      }
+    }
+
+    if !closeDelimiterOnNewLine {
+      unexpectedBeforeCloseQuote = [closeQuote]
+      closeQuote = RawTokenSyntax(missing: closeQuote.tokenKind, leadingTriviaPieces: [.newlines(1)], arena: self.arena)
+
+      // The closing delimiter doesn't start on a new line and thus it doesn't
+      // make sense to try and extract indentation from it.
+      return (unexpectedBeforeOpenQuote, openQuote, allSegments, unexpectedBeforeCloseQuote, closeQuote)
+    }
+
+    // -------------------------------------------------------------------------
+    // Parse indentation
+
+    if let lastSegment = lastSegment,
+      let parsedTrivia = parseIndentationTrivia(text: lastSegment.content.tokenText)
+    {
+      indentationTrivia = parsedTrivia
+      indentation = lastSegment.content.tokenText
+      closeQuote = closeQuote.extendingLeadingTrivia(by: parsedTrivia, arena: self.arena)
+    } else {
+      if let lastSegment = lastSegment {
+        indentationTrivia = TriviaParser.parseTrivia(lastSegment.content.tokenText, position: .leading).prefix(while: { $0.isIndentationWhitespace })
+        let indentationByteLength = indentationTrivia.reduce(0, { $0 + $1.byteLength })
+        indentation = SyntaxText(rebasing: lastSegment.content.tokenText[0..<indentationByteLength])
+        middleSegments.append(.stringSegment(lastSegment))
+      } else {
+        indentationTrivia = []
+        indentation = ""
+      }
+
+      unexpectedBeforeCloseQuote = [closeQuote]
+      closeQuote = RawTokenSyntax(missing: closeQuote.tokenKind, leadingTriviaPieces: [.newlines(1)] + indentationTrivia, arena: self.arena)
+    }
+
+    // -------------------------------------------------------------------------
+    // Check open quote followed by newline
+
+    // Condition for the loop below that indicates whether the segment we are
+    // iterating over is on a new line.
+    var isSegmentOnNewLine: Bool
+
+    if let newlineTrivia = firstSegment?.content.tokenText.triviaPieceIfNewline {
+      openQuote = openQuote.extendingTrailingTrivia(by: [newlineTrivia], arena: self.arena)
+      isSegmentOnNewLine = true
+    } else {
+      if let firstSegment = firstSegment {
+        middleSegments.insert(.stringSegment(firstSegment), at: 0)
```

Comment:
Following above this would then be `processed.append(.stringSegment(firstSegment))`.